rllib/policy/sample_batch.py

```diff
@@ -1245,7 +1245,12 @@ def wrap_as_needed(
             The single default policy's SampleBatch or a MultiAgentBatch
             (more than one policy).
         """
-        if len(policy_batches) == 1 and DEFAULT_POLICY_ID in policy_batches:
+        if (
+            len(policy_batches) == 1
+            and DEFAULT_POLICY_ID in policy_batches
+            # Otherwise: multi-agent case with DEFAULT_POLICY_ID
+            and policy_batches[DEFAULT_POLICY_ID].env_steps() == env_steps
```

Comment:
we are comparing these 2 env steps as a proxy for deciding whether we should build a MultiAgentBatch or SampleBatch.
there are only 2 places we call MultiAgentBatch.wrap_as_needed().
I wonder if we should be more explicit and actually figure out the right API, and also clean this up, so that we are not "guessing" anymore.

does this make sense? sounds like a good idea to you?